File: LeetCode/Python/Medium/442-Find-All-Duplicates-in-an-Array/solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        result=list()

        n=len(nums)

        for i in range(0, n):
            number=abs(nums[i])

            index=number - 1

            if nums[index] < 0:
                result.append(number)
            else:
                nums[index]*=-1

        return result

File: LeetCode/Rust/Medium/442-Find-All-Duplicates-in-an-Array/solution.rs
impl Solution
{
    pub fn find_duplicates(mut nums: Vec<i32>) -> Vec<i32>
    {
        let mut result: Vec<i32>=Vec::new();

        let n: usize=nums.len();

        for i in 0..n
        {
            let number: i32=nums[i].abs();

            let index: usize=(number - 1) as usize;

            if nums[index] < 0
            {
                result.push(number);
            }
            else
            {
                nums[index]*=-1;
            }
        }

        result
    }
}